util/slicesx/slicesx.go

```diff
@@ -42,3 +42,16 @@ func Shuffle[S ~[]T, T any](s S) {
 		s[i], s[j] = s[j], s[i]
 	}
 }
+
+// Partition returns two slices, the first containing the elements of the input
+// slice for which the callback evaluates to true, the second containing the rest.
+func Partition[S ~[]T, T any](s S, cb func(T) bool) (a, b S) {
```

Comment:
Side note: In the event you want a version that doesn't allocate (by permuting the slice in place), I wrote https://pkg.go.dev/github.com/creachadair/mds/slice#Partition for that.

Reply:
Oh neat! That's super handy to know about 👍